go/cmd/dolt/commands/fetch.go

```diff
@@ -179,19 +186,32 @@ func fetchRefSpecs(ctx context.Context, dEnv *env.DoltEnv, rem env.Remote, refSp
 			remoteTrackRef := rs.DestRef(branchRef)
 
 			if remoteTrackRef != nil {
-				verr := fetchRemoteBranch(ctx, dEnv, rem, srcDB, dEnv.DoltDB, branchRef, remoteTrackRef)
+				srcDBCommit, verr := fetchRemoteBranch(ctx, dEnv, rem, srcDB, dEnv.DoltDB, branchRef, remoteTrackRef)
 
 				if verr != nil {
 					return verr
 				}
+
+				if forceFetch {
+					err = dEnv.DoltDB.DeleteBranch(ctx, remoteTrackRef)
```

Comment:
How do we feel about this, versus just exposing a `SetHead` method? Seems like we're going to need/want that for an `update-ref` type functionality eventually anyway...and it would make this truly atomic.

```go
if forceFetch {
    err = dEnv.DoltDB.SetHead(ctx, remoteTrackRef, srcDBCommit)
} else {
    err = dEnv.DoltDB.FastForward(ctx, remoteTrackRef, srcDBCommit)
}
if err != nil {
    return errhand.BuildDError("error: fetch failed").AddCause(err).Build()
}
```